### PR TITLE
Implement hexadecimal floating point literals 

### LINKED ISSUE #326: Implement hexadecimal floating point literals
These are documented here:

https://ballerina.io/spec/lang/2021R1/#HexFloatingPointLiteral

They can be parsed with float:fromHexString.

### PR DESCRIPTION
**Pull request by ushirask**
Resolves #326

### CODE REVIEW

**Comment by ushirask** [comment]
Due to bug with `float:fromHexString` this will currently throw `error: invalid float literal` for hex float literals

File: compiler/modules/front.syntax/scan.bal [hunk 1]
```diff
@@ -642,40 +642,32 @@ function scanNumericEscape(int[] codePoints, int startIndex) returns int? {
...
-            |CP_UPPER_F
-            |CP_LOWER_A
-            |CP_LOWER_B
-            |CP_LOWER_C
-            |CP_LOWER_D
-            |CP_LOWER_E
-            |CP_LOWER_F => {
-                i += 1;
-                continue;
+    if i !=() {
```
Comment (jclark) [inline compiler/modules/front.syntax/scan.bal:648]: Use match.

File: compiler/modules/front.syntax/scan.bal [hunk 2]
```diff
@@ -764,6 +766,49 @@ function scanExponent(int[] codePoints, int startIndex) returns int? {
     return ();
 }
 
+function scanHexFractionExponent(int[] codePoints, int startIndex) returns int? {
```
Comment (jclark) [inline compiler/modules/front.syntax/scan.bal:769]: The common code between decimal and hex floating point should be factored out.

**Review by jclark — COMMENTED** [review]
Given that `float:fromHexString` isn't working yet, and given the amount of work that this patch still needs, I think you should stop working on this.

**Pull request closed without merging**